### ISSUE TITLE
How do I dismiss tabbar manually?

### ISSUE THREAD

**Issue by staticdreams** [picciano/TabBarReplacement#2]
I love your replacement for tabbar! It's awesome!

I'm having a small issue though. In my table I have lists of items (uicollectionview). By tapping one of each leads to a different view controller (view details of that item), where tabbar is no longer present (which is ok). 

When I tap on of those items, though, tabbar briefly shows up from the top, before I'm navigated to details. however, I'm tapping back button and coming back to my list of items tabbar is already present on the top. When I tap back button again (to bring tabbar down, it just goes further down, thinking it was dismissed previously). 

What I need to do is dismiss tabbar before navigating away from my list, so it won't be visible when I come back (last selected list item should be in focus when I'm back). Or maybe there is a way to prevent tabbar from showing up when I select of uicollection items and navigating away?

Let me illustrate this in screenshots to clarify things :)
1. Correct state (tabbar is in focus)
   ![01](https://cloud.githubusercontent.com/assets/988705/13546230/6b8dc37a-e2b8-11e5-933d-24ffae0f24b5.png)
2. One of the items is in focus now. Tabbar goes away. All good.
   ![02](https://cloud.githubusercontent.com/assets/988705/13546235/b85683fe-e2b8-11e5-8eb0-d69502984842.png)
3. When coming back from one of item view controller item is in focus, but tabbar should be invisible at this point.
   ![03](https://cloud.githubusercontent.com/assets/988705/13546238/da2062fc-e2b8-11e5-8739-eedeefce0505.png)
4. So, when calling tabbar, it just goes overlapping the main view controller
   ![04](https://cloud.githubusercontent.com/assets/988705/13546240/f8a8e9ba-e2b8-11e5-8336-5d5c3ab18869.png)

I hope it is clear enough what I'm trying to explain :) 
I'd try to fix this myself, but haven't figured out how your tabbarcontroller is wired exactly at this point..

Thanks!

**Comment by picciano** [comment]
First, be sure you are using the current version. I fixed a similar issue on February 10 and it might fix what you’re seeing too.
If that doesn’t fix it, please let me know how you are transitioning to the next view controller.

**Comment by staticdreams** [comment]
I was able to fix the issue. It turns out that when a new controller is pushed onto the one containing a tabbar, a method `viewDidLayoutSubviews` in `PITabBarController.swift` is always called, resetting the tabbar to it's initial state (even when it is not needed). So I fixed it like so:

```
override func viewDidLayoutSubviews() {
    if !tabBarHidden {
        _tabBar.frame = Constants.initialTabBarFrame
        _contentView.frame = self.view.bounds
    }
}
```

I can submit a PR if you want me to :)

However, there's lots to be done in searchviewcontroller, but I'll submit another ticket for that.
